src/compiler/checker.ts

```diff
@@ -46483,6 +46483,14 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         return undefined;
     }
 
+    function shouldParenthesizeConstantValue(node: AccessExpression) {
+        if (isAccessExpression(node.parent) && node.parent.expression === node) {
+            const value = getConstantValue(node);
+            return typeof value === "number" && value < 0;

```

Comment:
This should return `false` for `-Infinity`

Reply:
As far as I remember, using `Infinity/-Infinity` is [forbidden](https://www.typescriptlang.org/play?#code/MYewdgzgLgBApmArgWxgURgbwFAzzAQRgF4YBaASTADMBLMWqATwBpsBfbIA) for const enums.

Reply:
Yeah, the checker raises the error (which I didn't notice) but the emitter still handles that just OK:
```ts
// input
const enum TestEnum {
  A = -Infinity,
  B = Infinity,
}

console.log(TestEnum.A, TestEnum.B);

// output
"use strict";
console.log(-Infinity /* TestEnum.A */, Infinity /* TestEnum.B */);
```

Given that extra parenthesis shouldn't exactly hurt the `Infinity` case here, It's fine to ignore this here based on the fact that checker errors on this anyway 👍